scripts/billing/sql/daily_compute_usage_cud_sud_coverage.sql

```diff
@@ -0,0 +1,156 @@
+  FROM `committed-use-discount-test.billing.gcp_billing_export_v1_016676_F7EDE9_45E9C2`
+  WHERE service.description = "Compute Engine"
+  AND CAST(DATETIME(usage_start_time, "America/Los_Angeles") AS DATE) >= "2019-04-01"
+),
+
+-- create temporary table prices, in order to calculate unit price per (date, sku, region) tuple.
+-- Export table only includes the credit $ amount in the credit.amount field. We can get the credit
+-- usage amount (e.g. core hours) by dividing credit.amount ($) by unit price for that sku.
+-- This assumes that the unit price for the usage is equal to the unit price for the associated
+-- CUD credit. This should be correct, except in rare cases where unit price for that sku changed
+-- during the day (i.e. a price drop, change in spending-based discount %)
+-- It is necessary to do this in a separate table and join back into the main data set vs.
+-- separately on each usage line because some line items have CUD credit but no associated
+-- usage. We would not otherwise be able to calculate a unit price for these line items.
+prices AS (
+  SELECT  
+    usage_date,
+    sku.id AS sku_id,
+    location.region AS region,
+    -- calculate unit price per sku for each day. Catch line items with 0 usage to avoid divide by zero.
+    -- using 1 assumes that there are no relevant (CUD related) skus with cost but 0 usage, 
+    -- which is correct for current billing data
+    IF(SUM(usage.amount) = 0, 0, SUM(cost) / SUM(usage.amount)) as unit_price
+  FROM usage_data, UNNEST(credits) as cred
+  WHERE TRUE
+    AND cred.name LIKE "Committed%"
+    OR cred.name LIKE "Sustained%"
+  GROUP BY 1,2,3
+  ORDER BY 1,2,3
+),
+
+-- Temporary table to pull out CUD + SUD credits cost and usage data.
+-- Talculate usage amount (e.g. core*hrs) data by dividing costs by unit price.
+credit_data AS (
+  SELECT
+    usage_date,
+    region,
+    sku_id,
+    sku_description,
+    machine_spec,
+    project_id,
+    SUM(CUD_covered_usage) AS CUD_covered_usage,
+    SUM(CUD_cost) AS CUD_cost,
+    SUM(SUD_covered_usage) AS SUD_covered_usage,
+    SUM(SUD_cost) AS SUD_cost
+  FROM
+  (
+    SELECT
+      u.usage_date,
+      u.location.region AS region,
+      u.project.id AS project_id,
+      u.sku.id AS sku_id,
+      u.sku.description AS sku_description,
+      IFNULL(system_labels.value, "UNKNOWN") AS machine_spec,
+      usage.unit,
+      prices.unit_price,
+      cred.name,
+      IF (
+        prices.unit_price = 0 OR lower(cred.name) NOT LIKE "committed%",
+        0, 
+        CASE
+          -- Divide credit $ amount by unit price to calculate amount of usage offset by credit
+          WHEN LOWER(usage.unit) LIKE "seconds" THEN -1*SUM(cred.amount)/prices.unit_price      
+          WHEN LOWER(usage.unit) = "byte-seconds" THEN -1*SUM(cred.amount)/prices.unit_price
+          ELSE NULL
+        END
+      ) AS CUD_covered_usage,
+      IF (lower(cred.name) LIKE "committed%", SUM(cred.amount),0) AS CUD_cost,
```

Comment:
Capitalize `lower` to make consistent with the rest of the file.

Reply:
Done